sdcm/data_operations/data_operation.py

```diff
@@ -0,0 +1,293 @@
+        if rows and rows[0].count > 0:
+            raise ResurrectionDataFound(f"Found resurrection data in the table {self.schema_helpers.table_name}. "
+                                        f"Partition: %s", where_condition_to_delete)
+
+        for view in self.schema_helpers.view_names:
+            # Search for resurrection rows in the view
+            rows = fetch_all_rows(session=session, default_fetch_size=1000, statement=query % view, timeout=240)
+            self.log.debug("View %s has rows when filtered by '%s': %s", view, where_condition_to_delete, rows)
+            if rows and rows[0].count > 0:
+                raise ResurrectionDataFound(
+                    f"Found resurrection data in the view {view}. Partition: %s", where_condition_to_delete)
+
+    def validate_deleted_rows_not_resurrected(self, where_condition_to_delete):
+        with self.schema_helpers.cluster.cql_connection_patient(self.schema_helpers.cluster.nodes[0],
+                                                                keyspace=self.schema_helpers.keyspace_name) as session:
+            for row in where_condition_to_delete:
+                self.query_one_partition(where_condition_to_delete=row, session=session)
+
+    def count_rows(self, entity: str, limit: int = 1000):
+        try:
+            with self.schema_helpers.tester.db_cluster.cql_connection_patient(self.schema_helpers.tester.db_cluster.nodes[0],
+                                                                              keyspace=self.schema_helpers.keyspace_name) as session:
+                # Check if there are enough rows in the table or need to write more data
+                query = f"select count(*) as cnt from {entity} LIMIT {limit}"
+                self.log.debug("Run select query: %s", query)
+                result = self.schema_helpers.session_execute(session=session, query=query, timeout=300)
+                self.log.debug(
+                    "Rows in the %s.%s: %s", self.schema_helpers.keyspace_name, self.schema_helpers.table_name, result.one().cnt)
+                return result.one().cnt
+        except Exception as err:  # pylint: disable=broad-except
+            self.log.error("Failed query: %s. Error: %s", query, err)
+            return None
+
+    def delete_full_partition(self, partitions_amount_to_delete: int = 3):
+        with DataOperationsCaseEvent(case="DeleteFullPartition") as case_event:
+            try:
+                partition_values = self.get_unique_partition_values(partitions_amount=partitions_amount_to_delete)
+                where_condition = self.build_where_condition(columns=self.schema_helpers.partition_keys,
+                                                             values=partition_values)
+                self.log.debug("Where condition for delete: %s", where_condition)
+                self.delete_partition(where_condition_to_delete=where_condition)
+                time.sleep(self.interval)
```

Comment:
there multiple place we sleep the same interval, it's a bit confusing.
i.e. interval is used for 2 different things, sleeping between ops, and during the operation itself.